### PR TITLE
Added documentation for alternative docker build agent setup

### PR DESCRIPTION
**Pull request by daniels98it**
### Motivation and Context

After some difficulties getting jenkins working with the new pipeline builds, I came up with an alternative way of setting up a local build agent able to use the docker pipeline builds.

### Description

Added the documentation for an additional way of setting up a build agent using docker. The documentation describes how to set up an ssh build agent (similar to the remote build agents solution) using [jenkins/ssh-agent](https://hub.docker.com/r/jenkins/ssh-agent/) locally.

### CODE REVIEW

**Review by FuchsDominik — APPROVED** [review]
Text itself sounds good and couldn't find any errors :)

File: docs/dev/setup/jenkins-gitlab.rst [hunk 1]
```diff
@@ -733,6 +733,95 @@ Configure your master node like this  (adjust the number of executors, if needed
...
 
+Alternative local build agents setup using docker
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative way of adding a build agent that will use docker (similar to the remote agents below) but running
+locally, can be done using the jenkins/ssh-agent docker image (https://hub.docker.com/r/jenkins/ssh-agent).
+
+Prerequisites:
+
+1. Make sure to have Docker installed: https://docs.docker.com/engine/install/ubuntu/
```
Comment (FrankeLukas) [inline docs/dev/setup/jenkins-gitlab.rst:744]: Can we link to the more generic `https://docs.docker.com/engine/install/` here?

Comment (FrankeLukas) [reply docs/dev/setup/jenkins-gitlab.rst:744]: We also don't usually use plain links, I suggest the following:
```suggestion
1. Make sure to have Docker `installed <https://docs.docker.com/engine/install/ubuntu/>`__
```

File: docs/dev/setup/jenkins-gitlab.rst [hunk 2]
```diff
@@ -733,6 +733,95 @@ Configure your master node like this  (adjust the number of executors, if needed
 
       Jenkins local node
 
+Alternative local build agents setup using docker
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative way of adding a build agent that will use docker (similar to the remote agents below) but running
+locally, can be done using the jenkins/ssh-agent docker image (https://hub.docker.com/r/jenkins/ssh-agent).
```
Comment (FrankeLukas) [inline docs/dev/setup/jenkins-gitlab.rst:740]: Same here:
```suggestion
locally, can be done using the jenkins/ssh-agent `docker image <https://hub.docker.com/r/jenkins/ssh-agent>`__.
```

File: docs/dev/setup/jenkins-gitlab.rst [hunk 3]
```diff
@@ -733,6 +733,95 @@ Configure your master node like this  (adjust the number of executors, if needed
...
+
+1. Make sure to have Docker installed: https://docs.docker.com/engine/install/ubuntu/
+
+Agent setup:
+
+1. Create a new SSH key using ``ssh-keygen`` (if a passphrase is added, store it for later)
+
+2. Copy the public key content (e.g. in ~/.ssh/id_rsa.pub)
+
+3. Run ``docker run -d --name jenkins_agent -v /var/run/docker.sock:/var/run/docker.sock jenkins/ssh-agent:latest "<copied_public_key>"``
```
Comment (FrankeLukas) [inline docs/dev/setup/jenkins-gitlab.rst:752]: This command appears to be too long and doesn't render correctly:
grafik

File: docs/dev/setup/jenkins-gitlab.rst [hunk 4]
```diff
@@ -733,6 +733,95 @@ Configure your master node like this  (adjust the number of executors, if needed
...
+
+    ID: leave blank
+
+    Description: Up to you
+
+    Username: jenkins
+
+    Private Key: <content of the previous generated private key> (e.g /root/.ssh/id_rsa)
+
+    Passphrase: <the previous entered passphrase> (you can leave it blank if none has been specified)
```
Comment (FrankeLukas) [inline docs/dev/setup/jenkins-gitlab.rst:794]: This list as is looks odd to me:
grafik
Could we have a screenshot instead? (With the example values filled in.) Or perhaps a bullet point list/"code block"?

**Review by FrankeLukas — CHANGES_REQUESTED** [review]
Can't comment much on the technical side since I don't use this setup, but there are some minor visual issues we should have another look at.

**Comment by daniels98it** [comment]
Thanks @FrankeLukas for the suggestions. :rocket: 
These have been added with f46fefe and 98e51fd

**Review by FrankeLukas — APPROVED** [review]
Lgtm now. Also, I hope the RSA key you put in the screenshot is invalid by now. :sweat_smile:

**Comment by daniels98it** [comment]
> Lgtm now. Also, I hope the RSA key you put in the screenshot is invalid.

I copied it from [http://phpseclib.sourceforge.net/rsa/examples.html](http://phpseclib.sourceforge.net/rsa/examples.html) just to have an example for it.
So no worries :sweat_smile:

**Review by krusche — APPROVED** [review]

**Pull request merged**